### PR TITLE
Use api.app.EduIDApp to set up all our apps

### PR DESCRIPTION
**Pull request by johanlundberg**

### CODE REVIEW

File: src/eduid_common/api/app.py [hunk 1]
```diff
@@ -48,20 +50,72 @@
...
+        self.session_interface = SessionFactory(asdict(self.config))
+
+        if init_central_userdb:
+            self.central_userdb = UserDB(self.config.mongo_uri, 'eduid_am')
+
+        # Set up generic health check views
+        from eduid_common.api.views.status import status_views
+        self.register_blueprint(status_views)
+
+    def init_config(self, config_class, config):
```
Comment (enriquepablo) [inline src/eduid_common/api/app.py]: Not sure we still need this, in my memory I had removed it. I am testing it now in master, removing the method from `eduid_common.config.app` (and its execution in `eduid_common.api.app`) and it seems that it's only needed to allow checking for the `secret_key`.
mypy was being placated in each webapp, with the module from which current_app was being imported...

Comment (enriquepablo) [reply src/eduid_common/api/app.py]: Ignore this, sorry, there's more breakage. I agree that this should go, but it needs a bit more research.

File: src/eduid_common/api/app.py [hunk 2]
```diff
@@ -48,20 +50,72 @@
...
+        super(EduIDApp, self).__init__(name, **kwargs)
+        if config is None:
+            warnings.warn("config argument should be set to an app class config object",
+                          DeprecationWarning)
+            return
+        # Only try to load the key, value pairs that FlaskConfig expects
+        field_names = set(f.name for f in fields(FlaskConfig))
+        filtered_config = {k: v for k, v in config.items() if k in field_names}
+        # XXX: try except
+        self.config = FlaskConfig(**filtered_config)
```
[the remark this answers is not in the document]
Comment (johanlundberg) [reply src/eduid_common/api/app.py]: I see it as a feature to keep the common configuration separated from the specific app config.

File: src/eduid_common/api/app.py [hunk 3]
```diff
@@ -48,20 +50,72 @@
...
+        # Set app url prefix to APPLICATION_ROOT
+        self.wsgi_app = PrefixMiddleware(self.wsgi_app, prefix=self.config.application_root,  # type: ignore
+                                         server_name=self.config.server_name)
+
+        # Initialize shared features
+        init_logging(self)
+        init_exception_handlers(self)
+        init_sentry(self)
+        init_template_functions(self)
+        init_app_stats(self)
```
[the remark this answers is not in the document]
Comment (johanlundberg) [reply src/eduid_common/api/app.py]: I don't think they can be methods as they should be defined in the module that are responsible for initiating the actual feature. That said, there is probably nicer ways of doing this.

File: src/eduid_common/api/app.py (hunk 1 above)
Comment (johanlundberg) [reply src/eduid_common/api/app.py]: The init_config can go when all apps are instantiated in the [app_name]_init_app function I think.

**Comment by johanlundberg** [comment]
So do we want this?

**Pull request merged**